text_collector_examples/smartmon.sh

```diff
@@ -104,10 +112,15 @@ parse_smartctl_info() {
       esac
     fi
   done
-  echo "device_info{disk=\"${disk}\",type=\"${disk_type}\",vendor=\"${vendor}\",product=\"${product}\",revision=\"${revision}\",lun_id=\"${lun_id}\",model_family=\"${model_family}\",device_model=\"${device_model}\",serial_number=\"${serial_number}\",firmware_version=\"${fw_version}\"} 1"
-  echo "device_smart_available{disk=\"${disk}\",type=\"${disk_type}\"} ${smart_available}"
-  echo "device_smart_enabled{disk=\"${disk}\",type=\"${disk_type}\"} ${smart_enabled}"
-  echo "device_smart_healthy{disk=\"${disk}\",type=\"${disk_type}\"} ${smart_healthy}"
+  
+  if [ "$type" != "nvme" ]; then
+    echo "device_info{disk=\"${disk}\",type=\"${disk_type}\",vendor=\"${vendor}\",product=\"${product}\",revision=\"${revision}\",lun_id=\"${lun_id}\",model_family=\"${model_family}\",device_model=\"${device_model}\",serial_number=\"${serial_number}\",firmware_version=\"${fw_version}\"} 1"
+    echo "device_smart_available{disk=\"${disk}\",type=\"${disk_type}\"} ${smart_available}"
+    echo "device_smart_enabled{disk=\"${disk}\",type=\"${disk_type}\"} ${smart_enabled}"
+    echo "device_smart_healthy{disk=\"${disk}\",type=\"${disk_type}\"} ${smart_healthy}"
+  else
+    echo "nvme_device_info{disk=\"${disk}\",type=\"${disk_type}\",model_number=\"${model_number}\",serial_number=\"${serial_number}\",firmware_version=\"${fw_version}\"} 1"
```

Comment:
It seems odd to include 'nvme' in the metric name as the other metric names do not include an equivalent.

I'd suggest using `device_info` instead.

Reply:
Hmm, shouldn't all these metrics be `node_smart_...`  as they're SMART metrics about a node?

Reply:
I chose a different metric name as the label set is different from the standard device_info.